### PR TITLE
board: Set hostname from platform serial number

### PR DESCRIPTION
**Pull request by rajaganeshr**
Read serial number from chassis info area of HPM EEPROM set hostname as <platform name>-<serial num>

if HPM EEPROM is not programmed or does not have right format, then set hostname as <platform name>-<last 4 digits of mac addr>

Tested on Congo VRB and Qemu

### CODE REVIEW

**Review by ASADULLAHAMDGIT — APPROVED** [review]

**Review by supven01 — APPROVED** [review]

File: board/aspeed/evb_ast2700/evb_ast2700.c [hunk 1]
```diff
@@ -154,6 +162,50 @@ int get_platform_name( const u8 board_id, char* platname)
...
+	else
+		printf("invalid chassis serial number\n");
+	return csn_len;
+}
+
+int get_csn_uniq_str(const char* str, char delim, char* buf, size_t buf_len)
+{
+	int pos = 0;
+	int c;
+	for (c = 0; c < strlen(str); c++)
```
Comment (supven01) [inline board/aspeed/evb_ast2700/evb_ast2700.c]: strlen(str) should be outside of for loop, otherwise it will calculate strlen every iteration.

**Review by supven01 — APPROVED** [review]

File: board/aspeed/evb_ast2700/evb_ast2700.c (hunk 1 above)
Comment (rajaganeshr) [reply board/aspeed/evb_ast2700/evb_ast2700.c]: fixed in latest patchset

File: board/aspeed/evb_ast2700/evb_ast2700.c [hunk 2]
```diff
@@ -29,6 +29,14 @@
 #define HPM_BRD_ID_OFFSET    (18)
 #define HPM_BRD_REV_OFFSET   (19)
 #define STR_BUF_LEN         (128)
+#define FRU_HDR_OFF_MULTIPLIER  8
+#define CIA_START_OFFSET        2
+#define CIA_HDR_SIZE            3
+#define FRU_FIELD_TYPE_LEN_SIZE 1
+#define FRU_FIELD_TYPE_LEN_MASK 0x3F
+#define MIN_CSN_UNIQ_STR_LEN    4
+#define HYPEN_DELIM_SIZE        1
```
Comment (mahkurap) [inline board/aspeed/evb_ast2700/evb_ast2700.c]: nit: missing H in Hypen. It should be HYPHEN

**Review by mahkurap — CHANGES_REQUESTED** [review]

File: board/aspeed/evb_ast2700/evb_ast2700.c (hunk 2 above)
Comment (rajaganeshr) [reply board/aspeed/evb_ast2700/evb_ast2700.c]: Fixed

**Review by mahkurap — APPROVED** [review]

**Pull request merged**